### PR TITLE
Fixed apache.configfile to work with multiple same sections.

### PR DESCRIPTION
**Pull request by slivik**
### What does this PR do?
Fixes `apache.configfile` so that it allows to create Apache config files with multiple same section containers. It is possible to define structure like before (through a dict) and newly also through lists which allows sections with same name, e.g. `<Directory>`. Behavior is backward compatible and the new one is described in updated doc section in `apache` state module.

As side-effect it also fixes issue that when two structures were used at the same salt state run, the second structure was wrong since the previous run removed all `this:` items. So intended `<Directory /some/path>` was always generated as `<Directory>`.

### What issues does this PR fix or reference?
[#38306 - The `apache.configfile` state does not allow multiple same sections](https://github.com/saltstack/salt/issues/38306)

### Tests written?

Yes

### Commits signed with GPG?

Yes

### CODE REVIEW

File: salt/states/apache.py [hunk 1]
```diff
@@ -36,7 +36,55 @@
                   - Indexes
                   - FollowSymlinks
                 AllowOverride: All
+
+.. versionchanged:: 2018.3
```
Comment (rallytime) [inline salt/states/apache.py]: This should be `Fluorine`.

**Review by rallytime — COMMENTED** [review]
I realize you might be following what's already in the module, but is there some reason we're using `print` statements instead of `log`?

**Comment by gtmanfred** [comment]
Print is used to write to the config file. I will look at this tomorrow or Monday and review it.

**Review by gtmanfred — CHANGES_REQUESTED** [review]
I think this looks fine, but I would really like to see at least a unittest added for both this version and the old version to verify that we are not breaking backwards compatibility.

**Review by rallytime — APPROVED** [review]

**Review by gtmanfred — APPROVED** [review]

**Pull request merged**